src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmController.java

```diff
@@ -236,57 +203,54 @@ public CaseDocumentAmController(DocumentManagementService documentManagementServ
             ValidationService.isValidSecurityClassification(classification);
             ValidationService.validateLists(files, roles);
-            roles.forEach(role -> ValidationService.validateInputParams(INPUT_STRING_PATTERN, role));
 
             return documentManagementService.uploadDocuments(files, classification, roles,
-                                                             serviceAuthorization, caseTypeId, jurisdictionId, userId);
+                                                             caseTypeId, jurisdictionId, userId);
         } catch (BadRequestException | IllegalArgumentException e) {
+            LOG.error("Exception while uploading the documents :" + e);
             throw new BadRequestException("Exception while uploading the documents :" + e);
         } catch (Exception e) {
+            LOG.error("Exception while uploading the documents :" + e);
             throw new ResponseFormatException("Exception while uploading the documents :" + e);
         }
     }
 
-    @Override
+    //**************** Generate Hash Token API  API ***************
+
+    @ApiOperation(value = "Retrieves the hashcode for document Id")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Success", response = StoredDocumentHalResource.class)})
+    @RequestMapping(value = "/cases/documents/{documentId}/token", produces = {APPLICATION_JSON}, method = RequestMethod.GET)
     public ResponseEntity<Object> generateHashCode(
-        @ApiParam(value = Constants.S2S_API_PARAM, required = true)
-        @RequestHeader(value = Constants.SERVICE_AUTHORIZATION, required = true) String serviceAuthorization,
 
-        @ApiParam("Authorization header of the currently authenticated user")
-        @RequestHeader(value = "Authorization", required = true) String authorization,
+        @PathVariable("documentId") UUID documentId,
 
-        @ApiParam("documentId")
-        @PathVariable("documentId") UUID documentId) {
+        @NotNull(message = "Provide the Case Type ID ")
+        @RequestHeader(value = "caseTypeId", required = true) String caseTypeId,
 
+        @NotNull(message = "Provide the Jurisdiction ID ")
+        @RequestHeader(value = "jurisdictionId", required = true) String jurisdictionId) {
         try {
-            StoredDocumentHalResource resource = new StoredDocumentHalResource();
-            ResponseEntity responseEntity = documentManagementService.getDocumentMetadata(documentId);
-            if (responseEntity.getStatusCode().equals(HttpStatus.OK) && null != responseEntity.getBody()) {
-                resource = (StoredDocumentHalResource) responseEntity.getBody();
-            }
-
-            ValidationService.validateInputParams(INPUT_STRING_PATTERN, documentId.toString(),
-                                                  resource.getMetadata().get("caseTypeId"), resource.getMetadata().get("jurisdictionId"));
```

Comment:
why did we removed all this code? it seems your branch is not updated with latest development.

Reply:
Revert back again.Yes, you are right my branch was not updated.That's why it's seems as remove.Now,it's done.